pkg/service/http_service.go

```diff
@@ -111,14 +115,31 @@ func (s *HTTPService) Serve(ctx context.Context, eval eval.IEvaluator) error {
 	httpsl := tlsm.Match(cmux.HTTP1Fast())
 	gorpcl := tlsm.Match(cmux.Any())
 
-	go func() { handleServiceError(httpServer.Serve(httpl)) }() // HTTP
-
-	go func() { handleServiceError(httpServer.Serve(httpsl)) }() // HTTPS
-
-	go func() { handleServiceError(grpcServer.Serve(gorpcl)) }() // GRPC
-	go func() { handleServiceError(tlsm.Serve()) }()
-	go func() { handleServiceError(tcpm.Serve()) }()
-	<-ctx.Done()
+	g.Go(func() error {
```

Comment:
How do errors work here? 
If the first routine exits immediately does the error return get waited on until all return?
Or does the exit cause all other routines to exit?

Reply:
any error returned by one of the goroutines in the `errgroup` will cancel the group context will cancel the group context and `<-gCtx.Done()` will no longer block. The original error will be passed out from `g.Wait()`